src/services/chatgpt.ts

```diff
@@ -0,0 +1,88 @@
+const BASE_URL = 'https://api.openai.com/v1'
+const apiKey = '11111'
+
+interface EventDetails {
+  summary: string
+  location: string
+  start: {
+    dateTime: Date
+    timeZone: string
+  }
+  end: {
+    dateTime: Date
+    timeZone: string
+  }
+  description: string
+}
+
+type GetEventDetailsSuccessResponse = {
+  id: string
+  object: string
+  created: number
+  model: string
+  choices: {
+    message: {
+      role: string
+      content: string
+    }
+    finish_reason: string
+    index: number
+  }[]
+}
+
+type GetEventDetailsErrorResponse = {
+  isOk: false
+  data: null
+  error: {
+    message: string
+    type: string
+    param: null
+    code: string
+  }
+}
+
+type GetEventDetailsResponse = GetEventDetailsSuccessResponse | GetEventDetailsErrorResponse
+
+export async function getEventDetails(eventData: string): Promise<any> {
```

Comment:
Could be
```
export async function extractEventDetails(eventData: string): Promise<EventDetails | null> {
```

Reply:
changed the name of the function but the return type is problematic. it's a differently structured object for succuss and error response from the chatGPT server.